### PR TITLE
feat: seldon-core rock

### PR DESCRIPTION
**Pull request by i-chvets**
Summary of changes:
- Ported Selond Core rock.
- Added non-root user.
- Added security requirements.

### CODE REVIEW

File: seldon-core-operator/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    command: "/opt/seldon-core-operator/manager --enable-leader-election --webhook-port 4443 --create-resources true"
+platforms:
+  amd64:
+
+parts:
+  seldon-core-operator:
+    plugin: go
+    source: https://github.com/SeldonIO/seldon-core
+    source-type: git
+    source-tag: v1.15.1
```
Comment (misohu) [inline seldon-core-operator/rockcraft.yaml:31]: Why not 1.15.0 ? in main of seldon-operator we use 1.15.0

File: seldon-core-operator/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    build-snaps:
+      - go
+      - kustomize/latest/stable
```
Comment (misohu) [inline seldon-core-operator/rockcraft.yaml:34]: How about specifying version for go with 
```
build-snaps:
 - go/1.17/stable
```

File: seldon-core-operator/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      install -D generated/v1_service_seldon-webhook-service.yaml "${res}/service.yaml"
+      install -D generated/v1_configmap_seldon-config.yaml "${res}/configmap.yaml"
+      install -D \
+        generated/admissionregistration.k8s.io_v1_validatingwebhookconfiguration_seldon-validating-webhook-configuration.yaml \
+        "${res}/validate.yaml"
+      install -D \
+        generated/apiextensions.k8s.io_v1_customresourcedefinition_seldondeployments.machinelearning.seldon.io.yaml \
+        "${res}/crd-v1.yaml"
+
+      # security requirement
```
Comment (misohu) [inline seldon-core-operator/rockcraft.yaml:64]: Maybe we want to put it into separate part 

```
security-team-requirement:
    plugin: nil
    override-build: |
      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
```

File: seldon-core-operator/rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  non-root-user:
+    plugin: nil
+    after: [seldon-core-operator, mpl-deps]
+    overlay-script: |
+      # Create a user in the $CRAFT_OVERLAY chroot
+      groupadd -R $CRAFT_OVERLAY -g 1001 ubuntu
+      useradd -R $CRAFT_OVERLAY -M -r -u 1001 -g ubuntu ubuntu
+    override-prime: |
+      craftctl default
+      chown -R 1001:1001 opt/seldon-core-operator
```
Comment (misohu) [inline seldon-core-operator/rockcraft.yaml:114]: Are these two lines with `chown` necessary ? we are not doing it in other rocks

**Review by misohu — COMMENTED** [review]
Some comments. Good job.

File: seldon-core-operator/rockcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  Seldon core converts your ML models (Tensorflow, Pytorch, H2o, etc.) or
+  language wrappers (Python, Java, etc.) into production REST/GRPC
+  microservices.
+
+  Seldon handles scaling to thousands of production machine learning models and
+  provides advanced machine learning capabilities out of the box including
+  Advanced Metrics, Request Logging, Explainers, Outlier Detectors, A/B Tests,
+  Canaries and more.
+
+version: v1.15.1_22.04_1 # version format: <KF-upstream-version>_<base-version>_<Charmed-KF-version>
```
[the remark this answers is not in the document]
Comment (i-chvets) [reply seldon-core-operator/rockcraft.yaml:13]: Good point. Will change.

File: seldon-core-operator/rockcraft.yaml (hunk 2 above)
Comment (i-chvets) [reply seldon-core-operator/rockcraft.yaml:34]: If we do not specify version, latest go will be used which is v 1.20. It has less CVEs.

File: seldon-core-operator/rockcraft.yaml (hunk 3 above)
Comment (i-chvets) [reply seldon-core-operator/rockcraft.yaml:64]: Ok. Will do. We will need to address it in other rocks as well.

File: seldon-core-operator/rockcraft.yaml (hunk 4 above)
Comment (i-chvets) [reply seldon-core-operator/rockcraft.yaml:114]: May be we should, actually do it in other rocks as well.
Those are created as owned by root. Changing ownership plays into running as non-root story.

**Review by misohu — APPROVED** [review]

**Pull request merged**